src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\MainControls;
+
+use \ILIAS\UI\Component\Image\Image;
+
+/**
+ * This is what a factory for main controls looks like.
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Metabar is a unique page section to accomodate elements that
+	 *     should permamently be in sight of the user.
+	 *     The Metabar shall, first of all, host Prompts, i.e. notifications
```

Comment:
It would be nice to have this represented in the examples. Currently it uses bulky buttons only.